src/gluonts/model/deepstate/_estimator.py

```diff
@@ -0,0 +1,312 @@
+        state space model
+    past_length
+        This is the length of the training time series;
+        i.e., number of steps to unroll the RNN for before computing predictions.
+        Set this to (at most) the length of the shortest time series in the dataset.
+        (default: None, in which case the training length is set such that at least
+        `num_seasons_to_train` seasons are included in the training.
+        See `num_seasons_to_train`)
+    num_seasons_to_train
+        (Used only when `past_length` is not set)
+        Number of seasons to include in the training time series. (default: 4)
+        Here season corresponds to the longest cycle one can expect given the granularity of the time series.
+        See: https://stats.stackexchange.com/questions/120806/frequency-value-for-seconds-minutes-intervals-data-in-r
+    trainer
+        Trainer object to be used (default: Trainer())
+    num_layers
+        Number of RNN layers (default: 2)
+    num_cells
+        Number of RNN cells for each layer (default: 40)
+    cell_type
+        Type of recurrent cells to use (available: 'lstm' or 'gru';
+        default: 'lstm')
+    num_eval_samples
+        Number of samples paths to draw when computing predictions
+        (default: 100)
+    dropout_rate
+        Dropout regularization parameter (default: 0.1)
+    use_feat_dynamic_real
+        Whether to use the ``feat_dynamic_real`` field from the data
+        (default: False)
+    use_feat_static_cat
+        Whether to use the ``feat_static_cat`` field from the data
+        (default: False)
+    cardinality
+        Number of values of each categorical feature.
+        This must be set if ``use_feat_static_cat == True`` (default: None)
+    embedding_dimension
+        Dimension of the embeddings for categorical features (the same
+        dimension is used for all embeddings, default: 20)
+    scaling
+        Whether to automatically scale the target values (default: true)
+    time_features
+        Time features to use as inputs of the RNN (default: None, in which
+        case these are automatically determined based on freq)
+    """
+
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        add_trend: bool = False,
+        past_length: Optional[int] = None,
+        num_seasons_to_train: int = 4,
+        trainer: Trainer = Trainer(epochs=25, hybridize=False),
+        num_layers: int = 2,
+        num_cells: int = 40,
+        cell_type: str = "lstm",
+        num_eval_samples: int = 100,
+        dropout_rate: float = 0.1,
+        use_feat_dynamic_real: bool = False,
+        use_feat_static_cat: bool = False,
+        cardinality: Optional[List[int]] = None,
+        embedding_dimension: int = 20,
+        issm: Optional[ISSM] = None,
+        scaling: bool = True,
+        time_features: Optional[List[TimeFeature]] = None,
+    ) -> None:
+        super().__init__(trainer=trainer)
+
+        assert (
```

Comment:
You can use `prediction_length: pydantic.PositiveInt` and then remove the assert statements.